### ISSUE TITLE
Hanging indentation inside a multiline `if` condition

### ISSUE THREAD

**Issue by klmr** [r-lib/lintr#1942]
Inside an `if` condition that spans multiple lines, `indentation_linter` expects lines after the first line to be indented further.

Of course that’s an entirely valid stylistic choice but I’m wondering if there is a way to configure this linter such that the code below is permitted. I believe this is a fairly common convention, and I am not even sure whether the current behaviour is intentional.

(None of the values for `hanging_indent_style` changes the lint.)

```r
if (
  cond_a() ||
  cond_b()
) {}
```

‘lintr’ outputs:

> ```
> mwe.r:3:2: style: [indentation_linter] Indentation should be 4 spaces but is 2 spaces.
>  cond_b() ||
> ^~~
> ```

---

Incidentally I found that the following code is accepted by ‘lintr’, which arguably reads better than the above anyway. But on the other hand it no longer lines up the terms of the conditional.

```r
if (
  cond_1()
  || cond_2()
) {}
```

**Comment by klmr** [comment]
Another, potentially related, situation is the following piece of code:

```r
is_some_thing <- function() {
  condition_a() &&
    condition_b() && {
      vec = complex_prep_work()
      any(map_lgl(vec, some_predicate))
    }
}
```

The `indentation_linter` does *not* like this block. A workaround is to pull the block into its own function. But in the actual code it’s strictly single use, so the only purpose of putting it into a function would be to silence ‘lintr’.

**Comment by klmr** [comment]
* Related to #1718?

**Comment by klmr** [comment]
Apologies, one more issue with the linter, this time in RMarkdown. This might be better as a separate issue; I’ll leave it here for now but I’m happy to move it.

Consider the following RMarkdown document, which has fenced code inside an enumeration:

````r
* An enumeration item with code:

  ```{r}
  "Hello"
  ```
````

‘lintr’ has the spurious complaint

> ```
> mwe.rmd:4:2: style: [indentation_linter] Indentation should be 0 spaces but is 2 spaces.
>   "Hello"
> ~^
> ```

And in fact when removing the indent (which works, but is unconventional) and knitting the document, ‘knitr’ reinserts the indent in the output Markdown document.

**Comment by AshesITR** [comment]
Thanks for all the feedback on indentation linter.
Could you separate the problems into individual issues? That would make working on them and keeping track of them easier.

**Comment by klmr** [comment]
@AshesITR Done. Should I delete the comments above (or replace them with links to the relevant issues)?

**Comment by AshesITR** [comment]
Thanks! I think either way is fine.

**Comment by AshesITR** [comment]
NB `lintr` agrees with `styler` here:

``` r
styler::style_text("
if (
  cond_a() ||
  cond_b()
) {}
")
#> 
#> if (
#>   cond_a() ||
#>     cond_b()
#> ) {}
```

<sup>Created on 2023-04-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by bartekch** [comment]
I must admit that it's rather annoying that when using RStudio interactively (version 2023.06.0+421), the output version is not the one desired by `lintr` and `styler`, but rather the one from OP.
```r
if (
  cond_a() ||
  cond_b()
) {}
```
Surprisingly it works differently for a function call.
```r
foo(cond_a() ||
      cond_b())
```

I don't see anything specific about it in Tidyverse Style Guide. Are there any plans on dealing with this issue in `lintr`? Maybe it should be raised as an issue in RStudio?

**Comment by AshesITR** [comment]
The inconsistency seems worth filint with RStudio.
Regarding lintr support, this issue tracks it. Feel free to chip in with a PR if you like to work on it.

**Comment by bartekch** [comment]
FIY I finally filed an issue in RStudio regarding inconsistency in indentation - https://github.com/rstudio/rstudio/issues/13628
